### DISCUSSION TITLE
How to save multiple files using ui.upload() ?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1130] [Q&A] [answered]
### Question

Hi All,

I am using a ```ui.upload()``` where I am allowing user to upload multiple files (like excel, ppt, pdf, etc, just documents basically).

I went through the documentation to understand how to use the events and then get the ```name``` and ```content```. But I can't replicate it for multiple files. I am sure there is probably an easy solution to this, but I can't seem to figure it out. I tried to play around with the ```handle_upload``` function to try to make it into a list but could not make it.

Here is what I am doing now:
```python
def handle_upload(e: events.UploadEventArguments):
    filename = e.name
    filedata = e.content.read()
    savepath = os.path.join(sys.prefix,'uploads',filename)  # save the file in "uploads" folder
```

I thought if I can convert the function input as a list and process it, but I think this is wrong, does not work this way
```python
def handle_upload(files: List[events.UploadEventArguments]):
    for file in files:
        filename = file.name
        filedata = file.content.read()
```

Can you please suggest what would be the correct way to handle multiple files? 
Thanks a lot for your help.

Regards,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Hi @Anindya088!

The event handler `handle_upload` will be called for each uploaded file separately. You you only need to handle one file at a time. You can try it out with something like this:

```py
def handle_upload(e: events.UploadEventArguments):
    ui.notify(f'Uploaded {e.name}')

ui.upload(on_upload=handle_upload, multiple=True)
```

**Reply by Anindya088** [reply]
Hi @falkoschindler , 

Thanks so much for the reply, just now I tried and works perfectly :)
Basically this means I don't even have to modify the ```handle_upload``` function. If I am allowing ```multiple=True```, the function automatically repeats the job for all the uploaded files. This is so smooth 😄 

Thanks so much,
Anindya
